### DISCUSSION TITLE
[Bug] macOS non-native fullscreen mode renders behind MacBook notch

### DISCUSSION THREAD

**Discussion by jimeh** [ghostty-org/ghostty#3627] [Feature Requests, Ideas] [answered]
The non-native fullscreen mode is not aware of the notch area on recent MacBook machines when `macos-non-native-fullscreen` is set to `true`. When set to `visible-menu` is does render below the notch, but with the menu bar visible of course.

This was previously discussed in #378, but I'm not sure anything like `notch-padding` was ever added.

Essentially, on displays which feature a notch, the menu bar area should not be used to render content.

One current work-around is to manually add a top-padding by setting `window-padding-y` to `N,0`, where `N` matches the height of the notch. This height is different for different sized displays and resolutions however, so you'll have to try what work for your display+resolution combo.

**Comment by pluiedev** [comment, marked as answer]
Fixed in #5750
